src/ShallowWrapper.js

```diff
@@ -520,6 +521,29 @@ export default class ShallowWrapper {
   }
 
   /**
+   * Return a JSON representation of the node.
+   *
+   * NOTE: can only be called on a wrapper of a single node.
+   *
+   * @returns {Object}
+   */
+  json() {
+    return this.single('json', () => {
+      if (this.type() == null) {
+        return this.node;
+      }
+
+      const childrens = compact(this.children().map(c => c.json()));
+      return {
+        type: this.name(),
+        props: omit(this.props(), 'children'),
+        children: (childrens.length) ? childrens : null,
+        $$typeof: Symbol.for('react.test.json'),
+      };
```

Comment:
Should we warn if they're calling `json()` in an environment without `Symbol.for`? It's available in Node since `0.11`, and this wouldn't be very useful in a browser environment. That's another reason I think it should be called `snapshot`, since it's not meant to be used as a general-purpose `JSON` method.